src/Mint.js

```diff
@@ -0,0 +1,354 @@
+			setShowNote(true);
+			console.error(e);
+		}
+	}
+
+	const mintButton = saleStarted ? (
+		<>
+			<button
+				className="block w-full py-3 px-4 rounded-md shadow bg-indigo-500 text-white font-medium hover:bg-indigo-600 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-300 focus:ring-offset-gray-900"
+				onClick={() => mint(numTokensToPurchase)}
+			>
+				Mint
+			</button>
+			<p>
+				Mint {numTokensToPurchase} Faces for {(tokenPrice * numTokensToPurchase) / (10 ** 18)} ETH + GAS
+			</p>
+		</>
+	) : (
+		<button
+			className="opacity-50 block w-full py-3 px-4 rounded-md shadow bg-indigo-500 text-white font-medium"
+		>
+			Sale has not started	
+		</button>
+	)
+
+	return (
+		<div className="min-h-screen">
+			<div className="relative overflow-hidden">
+			<main>
+				<div className="pt-10 sm:pt-16 lg:pt-8 lg:pb-14 lg:overflow-hidden">
+					<div className="mx-auto max-w-7xl lg:px-8">
+						<div className="lg:grid lg:grid-cols-2 lg:gap-8">
+							<div className="mx-auto max-w-md px-4 sm:max-w-2xl sm:px-6 sm:text-center lg:px-0 lg:text-left lg:flex lg:items-center">
+								<div className="lg:py-24">
+									<h1 className="mt-4 text-4xl tracking-tight font-extrabold sm:mt-5 sm:text-6xl lg:mt-6 xl:text-6xl">
+										<span className="block">Mint a Face Token</span>
+									</h1>
+									<div className="mt-10 sm:mt-12 h-36">
+										{walletAddress ? (
+										<button
+											className="block w-full py-3 px-4 rounded-md shadow bg-green-500 text-white font-medium focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-300 focus:ring-offset-gray-900 mb-3"
+										>
+											Connected
+										</button>
+										) : (
+										<button
+											className="block w-full py-3 px-4 rounded-md shadow bg-indigo-500 text-white font-medium hover:bg-indigo-600 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-300 focus:ring-offset-gray-900 mb-3"
+											onClick={signIn}
+										>
+											Connect Wallet
+										</button>
+										)}
+										<div className="sm:flex">
+											{saleStarted && <div className="min-w-0 flex-1">
+												<label htmlFor="text" className="sr-only">
+													Number of tokens to mint
+												</label>
+												<div className="relative rounded-md">
+													<input
+														id="num-tokens"
+														type="text"
+														placeholder="Number of tokens to mint"
+														className="block w-full px-4 py-3 rounded-md border-0 text-base text-gray-900 placeholder-gray-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-300 focus:ring-offset-gray-900"
+														min="0"
+														max="20"
+														value={numTokensToPurchase}
+														onChange={ e => {
+															setNumTokensToPurchase(e.target.value || '');
+															setValidNumTokens(e.target.value > 0 && e.target.value < 21);
```

Comment:
not sure what this function is doing, but if the page default is 1, and they want to delete the 1 and type 20, will they be able to do so? some website have "validation" that doesn't allow basic typing

Reply:
@shivaalroy yes they can type in any number but if it's invalid then they will see a little error message. We can change the behavior if we think it's too noisy. What do you mean by not allow basic typing?